### PR TITLE
Create Prod Docker Container

### LINKED ISSUE #86: Update Docker Container
Update the docker containers on dockerhub such that they contain the code and can be ran independently of pulling any code from git.

### PR DESCRIPTION
**Pull request by Speuce**
## Tickets
Closes #86 #63 
## Summary Of Changes
- Created an all-in-one docker container for deploying to prod.
## Notes for Reviewers
~~1. Start the db container. DO NOT start any other containers.~~
~~2. From the base of the project run `docker compose -f docker-compose.prod.yml up`~~
~~3. Jamco should display on localhost:8000~~
~~4. Any changes made on the frontend should not propagate~~

**EDIT: ** You won't be able to run prod on local. It relies on HTTPS.. which requires an SSL cert. Just review it for code quality/etc. Also make sure I didn't break your local dev environment by setting up the containers again.

Also:
- I removed the test for jamco/views.py. It relies far to heavily on django setup which is impossible to mock out. I'd consider it more 'django boilerplate' than anything.

### CODE REVIEW

**Comment by jaredwebber** [comment]
Getting the following error when I run the docker-compose command. I got it with existing containers, and also after I deleted the JamCo containers used by development. I can look into it more later just wanted to make a note in case its not just a me problem.

```shell
jaredwebber@Jareds-MBP JAMCo % docker compose -f docker-compose.prod.yml up
[+] Running 2/2
 ⠿ Network jamco_default   Created                                                                                                        0.0s
 ⠿ Container jamco-prod-1  Created                                                                                                        0.1s
Attaching to jamco-prod-1
jamco-prod-1  | Starting django in 5 seconds...
jamco-prod-1  | Traceback (most recent call last):
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/base.py", line 282, in ensure_connection
jamco-prod-1  |     self.connect()
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/asyncio.py", line 26, in inner
jamco-prod-1  |     return func(*args, **kwargs)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/base.py", line 263, in connect
jamco-prod-1  |     self.connection = self.get_new_connection(conn_params)
jamco-prod-1  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/asyncio.py", line 26, in inner
jamco-prod-1  |     return func(*args, **kwargs)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/postgresql/base.py", line 215, in get_new_connection
jamco-prod-1  |     connection = Database.connect(**conn_params)
jamco-prod-1  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
jamco-prod-1  |     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  | psycopg2.OperationalError: could not translate host name "db" to address: Name does not resolve
jamco-prod-1  | 
jamco-prod-1  | 
jamco-prod-1  | The above exception was the direct cause of the following exception:
jamco-prod-1  | 
jamco-prod-1  | Traceback (most recent call last):
jamco-prod-1  |   File "/backend/manage.py", line 22, in <module>
jamco-prod-1  |     main()
jamco-prod-1  |   File "/backend/manage.py", line 18, in main
jamco-prod-1  |     execute_from_command_line(sys.argv)
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 446, in execute_from_command_line
jamco-prod-1  |     utility.execute()
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/__init__.py", line 440, in execute
jamco-prod-1  |     self.fetch_command(subcommand).run_from_argv(self.argv)
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 402, in run_from_argv
jamco-prod-1  |     self.execute(*args, **cmd_options)
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 448, in execute
jamco-prod-1  |     output = self.handle(*args, **options)
jamco-prod-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/base.py", line 96, in wrapped
jamco-prod-1  |     res = handle_func(*args, **kwargs)
jamco-prod-1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/core/management/commands/migrate.py", line 114, in handle
jamco-prod-1  |     executor = MigrationExecutor(connection, self.migration_progress_callback)
jamco-prod-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/migrations/executor.py", line 18, in __init__
jamco-prod-1  |     self.loader = MigrationLoader(self.connection)
jamco-prod-1  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/migrations/loader.py", line 58, in __init__
jamco-prod-1  |     self.build_graph()
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/migrations/loader.py", line 235, in build_graph
jamco-prod-1  |     self.applied_migrations = recorder.applied_migrations()
jamco-prod-1  |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/migrations/recorder.py", line 81, in applied_migrations
jamco-prod-1  |     if self.has_table():
jamco-prod-1  |        ^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/migrations/recorder.py", line 57, in has_table
jamco-prod-1  |     with self.connection.cursor() as cursor:
jamco-prod-1  |          ^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/asyncio.py", line 26, in inner
jamco-prod-1  |     return func(*args, **kwargs)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/base.py", line 323, in cursor
jamco-prod-1  |     return self._cursor()
jamco-prod-1  |            ^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/base.py", line 299, in _cursor
jamco-prod-1  |     self.ensure_connection()
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/asyncio.py", line 26, in inner
jamco-prod-1  |     return func(*args, **kwargs)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/base.py", line 281, in ensure_connection
jamco-prod-1  |     with self.wrap_database_errors:
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
jamco-prod-1  |     raise dj_exc_value.with_traceback(traceback) from exc_value
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/base.py", line 282, in ensure_connection
jamco-prod-1  |     self.connect()
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/asyncio.py", line 26, in inner
jamco-prod-1  |     return func(*args, **kwargs)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/base/base.py", line 263, in connect
jamco-prod-1  |     self.connection = self.get_new_connection(conn_params)
jamco-prod-1  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/utils/asyncio.py", line 26, in inner
jamco-prod-1  |     return func(*args, **kwargs)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/django/db/backends/postgresql/base.py", line 215, in get_new_connection
jamco-prod-1  |     connection = Database.connect(**conn_params)
jamco-prod-1  |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  |   File "/usr/local/lib/python3.11/site-packages/psycopg2/__init__.py", line 122, in connect
jamco-prod-1  |     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
jamco-prod-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
jamco-prod-1  | django.db.utils.OperationalError: could not translate host name "db" to address: Name does not resolve
```

**Review by LokiFrostGiant — COMMENTED** [review]
Is there no way to combine both backend/static and backend/global_static ? Currently both exist on this branch, so it might be worth looking into.

File: .platform/hooks/postdeploy/copy_staticfiles.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+docker cp $(docker ps -aqf "name=jamco"):/backend/static/ ./static/
+aws s3 rm --recursive  s3://jamco/static/
+aws s3 sync ./static/  s3://jamco/static/
```
Comment (jaredwebber) [inline .platform/hooks/postdeploy/copy_staticfiles.sh:4]:
```suggestion
aws s3 sync ./static/  s3://jamco/static/

```

**Review by jaredwebber — APPROVED** [review]
Just 1 nit, LGTM

**Pull request merged**